src/services/readingtip_service.py

```diff
@@ -14,6 +14,15 @@ def get_tips(self):
         else:
             return []
 
+    def change_tip(self, tip_object, title, link):
+        assert self._login_service.is_authenticated()
+        if tip_object.user == self._login_service.current_user():
+            if not link.startswith("http://") and not link.startswith("https://"):
+                link = "http://" + link
```

Comment:
For symmetry, this logic should probably be in the model, though I can shuffle it there when writing validations.